code/game/objects/items/storage/holsters.dm

```diff
@@ -0,0 +1,82 @@
+
+/obj/item/storage/belt/holster
+	name = "shoulder holster"
+	desc = "A rather plain but still badass looking holster."
+	icon_state = "holster"
+	item_state = "holster"
+	alternate_worn_layer = UNDER_SUIT_LAYER
+
+/obj/item/storage/belt/holster/equipped(mob/user, slot)
+	. = ..()
+	if(ishuman(user) && slot == ITEM_SLOT_BELT)
+		ADD_TRAIT(user, TRAIT_GUNFLIP, CLOTHING_TRAIT)
+
+/obj/item/storage/belt/holster/dropped(mob/user)
+	. = ..()
+	REMOVE_TRAIT(user, TRAIT_GUNFLIP, CLOTHING_TRAIT)
+
+/obj/item/storage/belt/holster/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.max_items = 1
+	STR.max_w_class = WEIGHT_CLASS_NORMAL
+	STR.set_holdable(list(
+		/obj/item/gun/ballistic/automatic/pistol,
+		/obj/item/gun/ballistic/revolver,
+		/obj/item/gun/energy/e_gun/mini,
+		/obj/item/gun/energy/disabler,
+		/obj/item/gun/energy/pulse/carbine,
+		/obj/item/gun/energy/dueling
+		))
+
+/obj/item/storage/belt/holster/detective
+	name = "detective's holster"
+	desc = "A holster to carry a handgun and ammo. WARNING: Badasses only."
+
+/obj/item/storage/belt/holster/detective/ComponentInitialize()
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.max_items = 3
+	STR.max_w_class = WEIGHT_CLASS_NORMAL
+	STR.set_holdable(list(
+		/obj/item/gun/ballistic/revolver,
+		/obj/item/ammo_box,
+		/obj/item/gun/energy/disabler,
+		/obj/item/gun/energy/dueling
```

Comment:
IMO, the detective's holster should be allowed to hold all of the kinds of things that normal holsters can (in addition to ammo).

Reply:
I disagree.